### PR TITLE
[ELY-957] Coverity static analysis: DefaultSingleSignOn.getIdentity() not synchronized

### PR DESCRIPTION
**Pull request by ivassile**
Wildfly Elytron: https://issues.jboss.org/browse/ELY-957

### CODE REVIEW

**Comment by darranl** [comment]
@pferraro would it be possible for you to please review this one?

From what I see the set is synchronized to ensure a first set wins.  Martin points out in the Jira issue that the current impl is thread safe but looking at the Javadoc for SingleSignOnEntry we don't mention the threading requirements.  

Also as a general comment there are actually multiple get methods - if synchronization is the correct answer we probably want it on all of them but as this is per-request I imagine unsynchronized access would be preferred.

**Comment by pferraro** [comment]
@darranl I'll have a look today.

File: src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOn.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public String getName() {
     }
 
     @Override
-    public SecurityIdentity getIdentity() {
+    public synchronized SecurityIdentity getIdentity() {
```
Comment (pferraro) [inline src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOn.java]: :-1: There's no reason to synchronize this method.  If the synchronized modifier in the method signature of DefaultSingleSignOn.setIdentity(...) is cumbersome for subclasses, then there are better ways to address this (e.g. by replacing the modifier with a synchronized block, and clarifying the javadoc).

**Comment by pferraro** [comment]
Superseded by https://github.com/wildfly-security/wildfly-elytron/pull/697

**Comment by ivassile** [comment]
Closing it since it is superseded by https://github.com/wildfly-security/wildfly-elytron/pull/697

**Pull request closed without merging**